pkg/volume/git_repo test has loopvar sharing bug

### What happened?

Apply this patch:

```
diff --git a/pkg/volume/git_repo/git_repo_test.go b/pkg/volume/git_repo/git_repo_test.go
index 3ef1cab0ad9..5e4806f152b 100644
--- a/pkg/volume/git_repo/git_repo_test.go
+++ b/pkg/volume/git_repo/git_repo_test.go
@@ -390,6 +390,7 @@ func doTestSetUp(scenario struct {
 	var fakeOutputs []fakeexec.FakeAction
 	var fcmd fakeexec.FakeCmd
 	for _, expected := range expecteds {
+		expected := expected
 		if expected.cmd[1] == "clone" {
 			fakeOutputs = append(fakeOutputs, func() ([]byte, []byte, error) {
 				// git clone, it creates new dir/files
```

Then run:

    make test WHAT=./pkg/volume/git_repo

The test fails:

```
+++ [0521 00:30:39] Setting GOMAXPROCS: 48
+++ [0521 00:30:39] Running tests without code coverage and with -race
--- FAIL: TestPlugin (0.01s)
    git_repo_test.go:277: Unexpected failure for scenario: target-dir - [unexpected command calls in scenario: expected 3, saw: 1 unexpected commands: [[git clone -- https://github.com/kubernetes/kubernetes.git target_dir]], expected: [[git clone -- https://github.com/kubernetes/kubernetes.git target_dir] [git checkout 2a30ce65c5ab586b98916d83385c5983edd353a1] [git reset --hard]] unexpected directories: [/tmp/git_repo_test.109792826/pods/poduid/volumes/kubernetes.io~git-repo/vol1], expected: [/tmp/git_repo_test.109792826/pods/poduid/volumes/kubernetes.io~git-repo/vol1 /tmp/git_repo_test.109792826/pods/poduid/volumes/kubernetes.io~git-repo/vol1/target_dir /tmp/git_repo_test.109792826/pods/poduid/volumes/kubernetes.io~git-repo/vol1/target_dir]]
    git_repo_test.go:277: Unexpected failure for scenario: no-target-dir - [unexpected command calls in scenario: expected 3, saw: 1 unexpected commands: [[git clone -- https://github.com/kubernetes/kubernetes.git]], expected: [[git clone -- https://github.com/kubernetes/kubernetes.git] [git checkout 2a30ce65c5ab586b98916d83385c5983edd353a1] [git reset --hard]] unexpected directories: [/tmp/git_repo_test.1084130303/pods/poduid/volumes/kubernetes.io~git-repo/vol1], expected: [/tmp/git_repo_test.1084130303/pods/poduid/volumes/kubernetes.io~git-repo/vol1 /tmp/git_repo_test.1084130303/pods/poduid/volumes/kubernetes.io~git-repo/vol1/kubernetes /tmp/git_repo_test.1084130303/pods/poduid/volumes/kubernetes.io~git-repo/vol1/kubernetes]]
FAIL
FAIL	k8s.io/kubernetes/pkg/volume/git_repo	0.172s
FAIL
make: *** [Makefile:184: test] Error 1
```


### What did you expect to happen?

Test should pass.

### How can we reproduce it (as minimally and precisely as possible)?

See above.


### Anything else we need to know?

The full loop is:

```
	for _, expected := range expecteds {
		if expected.cmd[1] == "clone" {
			fakeOutputs = append(fakeOutputs, func() ([]byte, []byte, error) {
				// git clone, it creates new dir/files
				os.MkdirAll(filepath.Join(fcmd.Dirs[0], expected.dir), 0750)
				return []byte{}, nil, nil
			})
		} else {
			// git checkout || git reset, they create nothing
			fakeOutputs = append(fakeOutputs, func() ([]byte, []byte, error) {
				return []byte{}, nil, nil
			})
		}
	}
```

The first problem is that fakeOutputs is capturing expected.dir, so all the funcs on that list use the expected.dir from the final iteration, not the one from the iteration with the expected.cmd[1] == "clone". But then this problem appears to be covering up a different problem in the test scenarios themselves. I looked at it for a little bit but I haven't puzzled out what the test is trying to do exactly. Hopefully someone with more knowledge of the test can take a look. 

It seems like a buggy test, accidentally passing due to this other loopvar sharing bug.



### Kubernetes version

HEAD

### Cloud provider

N/A

### OS version

N/A

### Install tools

N/A

### Container runtime (CRI) and version (if applicable)

N/A

### Related plugins (CNI, CSI, ...) and versions (if applicable)

N/A

--- a/pkg/volume/git_repo/git_repo_test.go
+++ b/pkg/volume/git_repo/git_repo_test.go
@@ -20,6 +20,7 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
+	"path"
 	"path/filepath"
 	"reflect"
 	"strings"
@@ -390,10 +391,20 @@ func doTestSetUp(scenario struct {
 	var fakeOutputs []fakeexec.FakeAction
 	var fcmd fakeexec.FakeCmd
 	for _, expected := range expecteds {
+		expected := expected
 		if expected.cmd[1] == "clone" {
+			// Calculate the subdirectory clone would create (if any)
+			// git clone -- https://github.com/kubernetes/kubernetes.git target_dir --> target_dir
+			// git clone -- https://github.com/kubernetes/kubernetes.git            --> kubernetes
+			// git clone -- https://github.com/kubernetes/kubernetes.git .          --> .
+			// git clone -- https://github.com/kubernetes/kubernetes.git ./.        --> .
+			cloneSubdir := path.Base(expected.cmd[len(expected.cmd)-1])
+			if cloneSubdir == "kubernetes.git" {
+				cloneSubdir = "kubernetes"
+			}
 			fakeOutputs = append(fakeOutputs, func() ([]byte, []byte, error) {
 				// git clone, it creates new dir/files
-				os.MkdirAll(filepath.Join(fcmd.Dirs[0], expected.dir), 0750)
+				os.MkdirAll(filepath.Join(fcmd.Dirs[0], expected.dir, cloneSubdir), 0750)
 				return []byte{}, nil, nil
 			})
 		} else {
@@ -422,7 +433,10 @@ func doTestSetUp(scenario struct {
 	g := mounter.(*gitRepoVolumeMounter)
 	g.exec = fake
 
-	g.SetUp(volume.MounterArgs{})
+	err := g.SetUp(volume.MounterArgs{})
+	if err != nil {
+		allErrs = append(allErrs, err)
+	}
 
 	if fake.CommandCalls != len(expecteds) {
 		allErrs = append(allErrs,

